recipes/folly/all/conanfile.py

```diff
@@ -132,78 +132,91 @@ def validate(self):
+            tc.variables["FOLLY_HAVE_UNALIGNED_ACCESS_EXITCODE"] = "0"
+            tc.variables["FOLLY_HAVE_UNALIGNED_ACCESS_EXITCODE__TRYRUN_OUTPUT"] = ""
+            tc.variables["FOLLY_HAVE_LINUX_VDSO_EXITCODE"] = "0"
+            tc.variables["FOLLY_HAVE_LINUX_VDSO_EXITCODE__TRYRUN_OUTPUT"] = ""
+            tc.variables["FOLLY_HAVE_WCHAR_SUPPORT_EXITCODE"] = "0"
+            tc.variables["FOLLY_HAVE_WCHAR_SUPPORT_EXITCODE__TRYRUN_OUTPUT"] = ""
+            tc.variables["HAVE_VSNPRINTF_ERRORS_EXITCODE"] = "0"
+            tc.variables["HAVE_VSNPRINTF_ERRORS_EXITCODE__TRYRUN_OUTPUT"] = ""
 
         if self.options.get_safe("use_sse4_2") and str(self.settings.arch) in ['x86', 'x86_64']:
-            # in folly, if simd >=sse4.2, we also needs -mfma flag to avoid compiling error.
+            tc.preprocessor_definitions["FOLLY_SSE"] = "4"
+            tc.preprocessor_definitions["FOLLY_SSE_MINOR"] = "2"
             if not is_msvc(self):
-                cmake.definitions["CMAKE_C_FLAGS"] = "-mfma"
-                cmake.definitions["CMAKE_CXX_FLAGS"] = "-mfma"
+                tc.variables["CMAKE_C_FLAGS"] = "-mfma"
+                tc.variables["CMAKE_CXX_FLAGS"] = "-mfma"
             else:
-                cmake.definitions["CMAKE_C_FLAGS"] = "/arch:FMA"
-                cmake.definitions["CMAKE_CXX_FLAGS"] = "/arch:FMA"
-
-        cmake.definitions["CMAKE_POSITION_INDEPENDENT_CODE"] = self.options.get_safe("fPIC", True)
-
-        cxx_std_flag = tools.cppstd_flag(self.settings)
-        cxx_std_value = cxx_std_flag.split('=')[1] if cxx_std_flag else "c++{}".format(self._minimum_cpp_standard)
-        cmake.definitions["CXX_STD"] = cxx_std_value
-        if is_msvc:
-            cmake.definitions["MSVC_LANGUAGE_VERSION"] = cxx_std_value
-            cmake.definitions["MSVC_ENABLE_ALL_WARNINGS"] = False
-            cmake.definitions["MSVC_USE_STATIC_RUNTIME"] = "MT" in msvc_runtime_flag(self)
-        cmake.configure()
-        return cmake
+                tc.variables["CMAKE_C_FLAGS"] = "/arch:FMA"
+                tc.variables["CMAKE_CXX_FLAGS"] = "/arch:FMA"
+
+        tc.variables["CMAKE_POSITION_INDEPENDENT_CODE"] = self.options.get_safe("fPIC", True)
+        # Relocatable shared lib on Macos
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0042"] = "NEW"
+        # Honor BUILD_SHARED_LIBS from conan_toolchain (see https://github.com/conan-io/conan/issues/11840)
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
+
+        cxx_std_value = self._cppstd_flag_value(self.settings.get_safe("compiler.cppstd", self._min_cppstd))
+        # 2019.10.21.00 -> either MSVC_ flags or CXX_STD
+        if is_msvc(self):
+            tc.variables["MSVC_LANGUAGE_VERSION"] = cxx_std_value
+            tc.variables["MSVC_ENABLE_ALL_WARNINGS"] = False
+            tc.variables["MSVC_USE_STATIC_RUNTIME"] = "MT" in msvc_runtime_flag(self)
+        else:
+            tc.variables["CXX_STD"] = cxx_std_value
```

Comment:
The old logic also set this so I think that should be fine

```suggestion
        tc.variables["CXX_STD"] = cxx_std_value
```

Reply:
I'm not sure CXX_STD is used when MSVC is in use:
https://github.com/facebook/folly/blob/main/CMakeLists.txt#L142
```
if(MSVC)
  include(FollyCompilerMSVC)
else()
  include(FollyCompilerUnix)
endif()
```